odo/backends/aws.py

```diff
@@ -219,15 +220,44 @@ def anything_to_s3_text(s3, o, **kwargs):
     return append(s3, convert(Temp(s3.subtype), o, **kwargs), **kwargs)
 
 
+@contextmanager
+def start_multipart_upload_operation(s3):
+    try:
+        multipart_upload = s3.object.bucket.initiate_multipart_upload(s3.key)
+        yield multipart_upload
+    except Exception as exception:
+        multipart_upload.cancel_multipart_upload()
+        raise exception
+    else:
+        multipart_upload.complete_upload()
+
+
 @append.register(S3(JSONLines), (JSONLines, Temp(JSONLines)))
 @append.register(S3(JSON), (JSON, Temp(JSON)))
 @append.register(S3(CSV), (CSV, Temp(CSV)))
 @append.register(S3(TextFile), (TextFile, Temp(TextFile)))
-def append_text_to_s3(s3, data, **kwargs):
+def append_text_to_s3(s3, data, multipart=False, **kwargs):
+    if multipart:
+        with start_multipart_upload_operation(s3) as multipart_upload:
+            _upload(data, multipart_upload, **kwargs)
+        return s3
+
     s3.object.set_contents_from_filename(data.path)
     return s3
 
 
+def _upload(data, multipart_upload, part_size=5 * 1 << 20, **kwargs):
+    parts_counter = 0
+
+    def upload_part(part):
+        multipart_upload.upload_part_from_file(BytesIO(part), part_num=parts_counter + 1)
+        return parts_counter + 1
+
+    with open(data.path, 'rb') as f:
+        for part in iter(curry(f.read, part_size), ''):
```

Comment:
i think this can be simplified even further:

```python
for part_num, part in enumerate(iter(curry(f.read, part_size), '')):
    upload_part(part, part_num=part_num)
```
then you wouldn't need this somewhat complex scope logic to increment the parts_counter